packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ArtifactController.java

```diff
@@ -210,12 +193,9 @@ public ResponseEntity<PresignedURL> getDownloadURL(
 			@RequestParam("filename") final String filename) {
 
 		try {
-			Optional<PresignedURL> url = artifactService.getDownloadUrl(id, filename);
-			if (url.isEmpty()) {
-				return ResponseEntity.notFound().build();
-			}
-			return ResponseEntity.ok(url.get());
-		} catch (Exception e) {
+			final Optional<PresignedURL> url = artifactService.getDownloadUrl(id, filename);
+			return url.map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.notFound().build());
```

Comment:
`notFound` returns a 404... you will need to update the ApiResponse

Reply:
fixed